### PR TITLE
Dummy PR for review

### PR DESCRIPTION
**Pull request by oppiliappan**
[Rendered readme](https://github.com/ngi-nix/openpgp-interoperability-test-suite/blob/dummy-target/readme.md).

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,92 @@
+{
+  description = "OpenPGP Interoperability Test Suite";
+
+  inputs = {
+
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-21.05";
+
+    # nettle-sys crate seems to use a custom build.rs script that breaks
+    # rustPlatform.buildRustPackage, works fine with importCargo
+    import-cargo.url = "github:edolstra/import-cargo";
```
Comment (happysalada) [inline flake.nix]: import-cargo is old and not really maintained if I'm not wrong.
with the buildRustPackage, what was the problem ?
There are two ways to use buildRustPackage, providing a vendor sha256 and passing the lockfile, did you try both ?
curious as to what was the problem with nettle-sys, someone else might have already figured out the solution in nixpkgs.

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            rustc
+            cargo
+            pkg-config
+            nettle
+            clang
+            llvm
+          ];
+
+          # required by custom build.rs in the nettle-sys crate
+          LIBCLANG_PATH = "${llvmPackages.libclang.lib}/lib";
```
Comment (happysalada) [inline flake.nix]: What is nettle-sys building ?
There might be an escape hatch to rely on a provided version of the tool.
openssl-sys has OPEN_SSL_NO_VENDOR for that.

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+              --replace 'templates/**/*' "${sequoia-src}/templates/**/*"
+          '';
+
+          buildPhase = ''
+            cargo build --release --offline
+          '';
+
+          installPhase = ''
+            mkdir -p $out/bin
+            install -Dm775 ./target/release/openpgp-interoperability-test-suite $out/bin/
```
Comment (happysalada) [inline flake.nix]: Just wondering.
How many implementations are there really ?
If it's less than 10, maybe we can wrap the resulting binary to enable for quick testing ?
(not sure this makes sense, just thinking out loud)

File: flake.nix (hunk 1 above)
Comment (oppiliappan) [reply flake.nix]: >providing a vendor sha256 and passing the lockfile, did you try both ?

I actually haven't tried the lockfile method, I will give that a go.

File: flake.nix (hunk 2 above)
Comment (oppiliappan) [reply flake.nix]: [nettle-sys](https://gitlab.com/sequoia-pgp/nettle-sys/-/blob/main/build.rs) seems to use a custom `build.rs` file, from the logs, what I could tell was this build script failed when built with `rustPlatform`.

File: flake.nix (hunk 3 above)
Comment (oppiliappan) [reply flake.nix]: There are [quite a few](https://wiki.gnupg.org/OtherFreeSoftwareOpenPGP) implementations, in varying languages. I don't really know what the specific use cases of this package are to be honest, so I am not sure if we should package a couple of implementations alongside.

Comment (happysalada) [reply flake.nix]: that works for me, was just trying to think out loud.

**Comment by leenaars** [comment]
@happysalada, @NerdyPepper: To work with the test suite, the implementation must have the Stateless OpenPGP Command Line Interface. That list is quite shorter than the list of all OpenPGP implementations, and very likely the shortlist of candidates coincides with the tested implementations here:

https://tests.sequoia-pgp.org

> so I am not sure if we should package a couple of implementations alongside.

If it is feasible, yes. This would help the test designers a lot, I understand, as an authoritative, reproducible test instance. Several of these are not in nixpkgs yet, would also for users indeed be great if they were.

**Comment by oppiliappan** [comment]
@leenaars Ah I see. In that case, we can package those alongside. I'll take a look at that.

File: flake.nix (hunk 1 above)
Comment (oppiliappan) [reply flake.nix]: Reworked to use `crate2nix` now.

**Comment by oppiliappan** [comment]
Will merge this, I will add implementations in subsequent PRs.

**Pull request closed without merging**